### ISSUE TITLE
Dealing with badly formatted files

### ISSUE THREAD

**Issue by valentinwust** [anders-biostat/metdense#4]
What is the best behaviour for the maker if the files are not ordered properly? There seem to be some small files in the scnmt_gastrulation data where the order is messed up, e.g. `E7.5_Plate2_G2` goes 1, 11, ..., 17, 2, ..., 9, 10, 18, X. How expensive would checking for this and dropping the relevant files be? Currently it goes through it in order and aborts if something is wrong. But working for three hours and aborting because of a minor problem like this doesnt seem optimal.

**Comment by simon-anders** [comment]
Yesterday evening, I've written two small utilities (commit #9d9f95) to find these badly sorted files and sort them. 

I've called them using some bash scripting to get a clean metdense file of the full gastrulation data set. I can upload it somewhere.

Eventually, we should come up with a better solution, but I'd leave it at that for now.

**Comment by simon-anders** [comment]
For convenience, the badly sorted files are the following files (which are all short):

```E4.5-5.5_new_Plate1_E09.tsv.gz 
E4.5-5.5_new_Plate2_D11.tsv.gz 
E4.5-5.5_new_Plate3_A07.tsv.gz 
E4.5-5.5_new_Plate3_H05.tsv.gz 
E4.5-5.5_new_Plate4_D11.tsv.gz 
E4.5-5.5_new_Plate4_F08.tsv.gz 
E6.5_Plate2_G6.tsv.gz 
E6.5_Plate3_B11.tsv.gz 
E6.5_Plate3_B2.tsv.gz 
E6.5_Plate3_F2.tsv.gz 
E6.5_Plate3_G2.tsv.gz 
E7.5_Plate11_E06.tsv.gz 
E7.5_Plate11_F03.tsv.gz 
E7.5_Plate11_G09.tsv.gz 
E7.5_Plate14_A10.tsv.gz 
E7.5_Plate14_D09.tsv.gz 
E7.5_Plate16_E08.tsv.gz 
E7.5_Plate2_G2.tsv.gz 
E7.5_Plate2_G3.tsv.gz 
E7.5_Plate4_C2.tsv.gz 
E7.5_Plate4_D2.tsv.gz 
E7.5_Plate4_G2.tsv.gz 
```

EDIT: The badly formatted files for the accessibility data are:
```
E4.5-5.5_new_Plate2_D11.tsv.gz 
E4.5-5.5_new_Plate3_A07.tsv.gz 
E4.5-5.5_new_Plate4_D11.tsv.gz 
E6.5_Plate3_B2.tsv.gz 
E6.5_Plate3_G2.tsv.gz 
E7.5_Plate11_B04.tsv.gz
E7.5_Plate14_D09.tsv.gz 
E7.5_Plate16_E08.tsv.gz
E7.5_Plate4_D2.tsv.gz
```

**Comment by valentinwust** [comment]
Great, that should be good enough for now. Is the full .metdense file somewhere on papagei? I can create my own, but it takes a few hours.

**Comment by simon-anders** [comment]
Uploading it at the moment. Will still take 45 min

**Comment by simon-anders** [comment]
It's now on papagei at /mnt/raid/anders/tmp/gastrulation.metdense.gz.

Note that due to a mistake on my part, the unsorted cells listed above are all empty in that file.

**Comment by valentinwust** [comment]
Something seems to have gone wrong, the data block of the file starts at 20522, which is not divisible by four. I could adjust my code to deal with that, but the maker shouldn't produce files that don't conform to the specification. I don't see how this could happen from the code.

**Comment by simon-anders** [comment]
> Something seems to have gone wrong, the data block of the file starts at 20522, which is not divisible by four. I could adjust my code to deal with that, but the maker shouldn't produce files that don't conform to the specification. I don't see how this could happen from the code.

fixed in 9f7843bb477b164d4b9102580a395ed38fd46652
